kong/router.lua

```diff
@@ -770,8 +774,7 @@ do
   local matchers = {
     [MATCH_RULES.HOST] = function(route_t, ctx)
       local req_host = ctx.hits.host or ctx.req_host
-      local host = route_t.hosts[req_host]
-        or route_t.hosts[split_port(req_host)]
+      local host = route_t.hosts[req_host] or route_t.hosts[ctx.host_no_port]
```

Comment:
This was my last remaining concern: I removed the call to `split_port` on this hot path, without any issues so far storing `host_no_port` in the context.